app/src/main/java/com/ottfstudio/quizdroid/presentation/quiz/QuizViewModel.kt

```
@@ -99,20 +101,23 @@ class QuizViewModel
                     recordRepository.insertQuizRecord(
                         Record(
                             date = state.value.today,
                             quiz = state.value.quiz,
-                            selectedOption = state.value.selectedOption ?: -1,
-                            isCorrect = state.value.quiz.answer == state.value.selectedOption,
+                            selectedOption = selectedOption,
+                            isCorrect = state.value.quiz.answer == selectedOption + 1,
                             isSolved = state.value.isSolved,
                         ),
                     )
                 } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)